src/server/routes/entity/entity.tsx

```diff
@@ -1201,11 +1203,21 @@ export function handleCreateOrEditEntity(
 	const {orm}: {orm?: any} = req.app.locals;
 	const editorJSON = req.user;
 	const {bookshelf} = orm;
-	const entityEditPromise = bookshelf.transaction((transacting) =>
-		processSingleEntity(req.body, res.locals.entity, req.session, entityType, orm, editorJSON, derivedProps, isMergeOperation, transacting));
-	const achievementPromise = entityEditPromise.then(
-		(entityJSON) => processAchievement(orm, editorJSON.id, entityJSON)
-	);
+	const entityEditPromise = bookshelf.transaction((transacting) => {
+		processSingleEntity(
+			req.body,
+			res.locals.entity,
+			req.session,
+			entityType,
+			orm,
+			editorJSON,
+			derivedProps,
+			isMergeOperation,
+			transacting
+		);
+	});
+	const entityJSON = await entityEditPromise;
```

Comment:
This does work, but why not the simpler `const entityJSON = await bookshelf.transaction(…` ?